packages/amplify_datastore/example/integration_test/separate_integration_tests/multi_relationship_test.dart

```diff
@@ -0,0 +1,443 @@
+    });
+
+    testWidgets('save meetings', (WidgetTester tester) async {
+      for (var meeting in meetings) {
+        if (enableCloudSync) {
+          var eventGetter = getExpectedSubscriptionDataProcessedEvent(
+            eventMatcher: (event) {
+              var model = event.element.model;
+              if (model is MultiRelatedMeeting) {
+                return model.id == meeting.id && event.element.version == 1;
+              }
+
+              return false;
+            },
+          );
+
+          await Amplify.DataStore.save(meeting);
+
+          var event = await eventGetter;
+          expect(event.element.deleted, isFalse);
+        } else {
+          await Amplify.DataStore.save(meeting);
+        }
+      }
+      var queriedMeetings =
+          await Amplify.DataStore.query(MultiRelatedMeeting.classType);
+      expect(queriedMeetings, isNotEmpty);
+    });
+
+    testWidgets('save attendees', (WidgetTester tester) async {
+      for (var attendee in attendees) {
+        if (enableCloudSync) {
+          var eventGetter = getExpectedSubscriptionDataProcessedEvent(
+            eventMatcher: (event) {
+              var model = event.element.model;
+              if (model is MultiRelatedAttendee) {
+                return model.id == attendee.id && event.element.version == 1;
+              }
+
+              return false;
+            },
+          );
+
+          await Amplify.DataStore.save(attendee);
+
+          var event = await eventGetter;
+          expect(event.element.deleted, isFalse);
+        } else {
+          await Amplify.DataStore.save(attendee);
+        }
+      }
+      var queriedAttendees =
+          await Amplify.DataStore.query(MultiRelatedAttendee.classType);
+      expect(queriedAttendees, isNotEmpty);
+    });
+
+    testWidgets('save registrations', (WidgetTester tester) async {
+      for (var registration in registrations) {
+        if (enableCloudSync) {
+          var eventGetter = getExpectedSubscriptionDataProcessedEvent(
+            eventMatcher: (event) {
+              var model = event.element.model;
+              if (model is MultiRelatedRegistration) {
+                return model.id == registration.id &&
+                    event.element.version == 1;
+              }
+
+              return false;
+            },
+          );
+
+          await Amplify.DataStore.save(registration);
+
+          var event = await eventGetter;
+          expect(event.element.deleted, isFalse);
+        } else {
+          await Amplify.DataStore.save(registration);
+        }
+      }
+      var queriedRegistrations =
+          await Amplify.DataStore.query(MultiRelatedRegistration.classType);
+      expect(queriedRegistrations, isNotEmpty);
+    });
+
+    testWidgets('query meetings', (WidgetTester tester) async {
+      var queriedMeetings =
+          await Amplify.DataStore.query(MultiRelatedMeeting.classType);
+      for (var meeting in queriedMeetings) {
```

Comment:
nit https://api.flutter.dev/flutter/package-matcher_matcher/containsAll.html, same for similar tests